### DISCUSSION TITLE
How to auto organize imports on save?

### DISCUSSION THREAD

**Discussion by abawchen** [nvim-java/nvim-java#200] [Q&A]
Does anyone know how to config to do that? Thanks.

**Comment by abawchen** [comment]
I am currently using following config, is there a better way?
```lua
vim.api.nvim_create_autocmd("BufWritePre", {
  callback = function()
    if vim.bo.ft == "java" then
      vim.lsp.buf.code_action({
        context = { only = { "source.organizeImports" } },
        apply = true,
      })
    end
  end,
})
```

**Reply by s1n7ax** [reply]
From what I can see, vscode is sending a `java/organizeImports` method request which is a little bit different

https://github.com/redhat-developer/vscode-java/blob/8e71cb28ccad18d35b20c682ae8cc869e178da74/src/protocol.ts?plain=1#L233

But it's in the specification. I think what you are doing is better and the standard.

**Reply by s1n7ax** [reply]
@abawchen added this
https://github.com/nvim-java/nvim-java/wiki/Tips-&-Tricks#running-code-actions
